src/main/java/io/github/bobocodebreskul/context/scan/utils/ScanUtilsImpl.java

```diff
@@ -60,6 +61,30 @@ public Set<Class<?>> searchClassesByFilter(String packagePath, Predicate<Class<?
   }
 
 
+  @Override
+  public String[] readBasePackages(Class<?> annotatedClass) {
+    BringComponentScan bringComponentScan = annotatedClass.getAnnotation(BringComponentScan.class);
+    if (bringComponentScan != null) {
+      if (bringComponentScan.basePackages().length != 0) {
+        String[] basePackages = bringComponentScan.basePackages();
+        log.info("Base packages read from @BringComponentScan annotation: " + String.join(", ",
+            basePackages));
+        return basePackages;
+      } else {
+        String basePackage = annotatedClass.getPackage().getName();
+        log.info("Using default base package: " + basePackage);
+        return new String[]{basePackage};
+      }
+    } else {
+      String errorMessage =
+          "No @BringComponentScan annotation found on class: " + annotatedClass.getSimpleName()
+              + ".class";
+      log.warn(errorMessage);
+      throw new BringComponentScanNotFoundException(errorMessage);
```

Comment:
You can remove additional if-else nesting by simply move this check to the very top of this method, it will loo like this:
```
if (bringComponentScan == null) {
  String errorMessage =
      "No @BringComponentScan annotation found on class: " + annotatedClass.getSimpleName()
          + ".class";
  log.warn(errorMessage);
  throw new BringComponentScanNotFoundException(errorMessage);
}
if (bringComponentScan.basePackages().length != 0) {
  String[] basePackages = bringComponentScan.basePackages();
  log.info("Base packages read from @BringComponentScan annotation: " + String.join(", ",
          basePackages));
...
```